patients/repo.go

```diff
@@ -931,3 +936,233 @@ func cmpToMongoFilter(cmp *string) (string, bool) {
+			"summary.cgmStats.dates.lastUploadDate": bson.M{
+				"$gt":  params.CgmLastUploadDateFrom,
+				"$lte": params.CgmLastUploadDateTo,
+			},
+			"summary.cgmStats.periods." + *params.Period + "." + category.Field: bson.M{category.Comparison: category.Value},
+		}
+
+		opts := options.Find()
+		opts.SetLimit(int64(limit))
+
+		if category.Comparison == "$gt" {
+			opts.SetSort(bson.D{
+				{"summary.cgmStats.periods." + *params.Period + "." + category.Field, -1},
+			})
+		} else {
+			opts.SetSort(bson.D{
+				{"summary.cgmStats.periods." + *params.Period + "." + category.Field, 1},
+			})
+		}
+
+		cursor, err := r.collection.Find(ctx, selector, opts)
+		if err != nil {
+			return nil, err
+		}
+
+		var patientsList []*Patient
+		if err = cursor.All(ctx, &patientsList); err != nil {
+			return nil, fmt.Errorf("error decoding patients list: %w", err)
+		}
+
+		categoryResult := make([]TideResultPatient, 0, 25)
+		for _, patient := range patientsList {
+			exclusions = append(exclusions, patient.Id)
+
+			var patientTags []string
+			for _, tag := range *patient.Tags {
+				patientTags = append(patientTags, tag.Hex())
+			}
+
+			categoryResult = append(categoryResult, TideResultPatient{
+				Patient: &TidePatient{
+					Email:    patient.Email,
+					FullName: patient.FullName,
+					Id:       patient.UserId,
+					Tags:     &patientTags,
+				},
+				AverageGlucoseMmol:         (*patient.Summary.CGM.Periods)[*params.Period].AverageGlucoseMmol,
+				GlucoseManagementIndicator: (*patient.Summary.CGM.Periods)[*params.Period].GlucoseManagementIndicator,
+				TimeCGMUseMinutes:          (*patient.Summary.CGM.Periods)[*params.Period].TimeCGMUseMinutes,
+				TimeCGMUsePercent:          (*patient.Summary.CGM.Periods)[*params.Period].TimeCGMUsePercent,
+				TimeInHighPercent:          (*patient.Summary.CGM.Periods)[*params.Period].TimeInHighPercent,
+				TimeInLowPercent:           (*patient.Summary.CGM.Periods)[*params.Period].TimeInLowPercent,
+				TimeInTargetPercent:        (*patient.Summary.CGM.Periods)[*params.Period].TimeInTargetPercent,
+				TimeInTargetPercentDelta:   (*patient.Summary.CGM.Periods)[*params.Period].TimeInTargetPercentDelta,
+				TimeInVeryHighPercent:      (*patient.Summary.CGM.Periods)[*params.Period].TimeInVeryHighPercent,
+				TimeInVeryLowPercent:       (*patient.Summary.CGM.Periods)[*params.Period].TimeInVeryLowPercent,
+			})
+		}
+
+		(*tide.Results)[category.Heading] = &categoryResult
+
+		limit -= len(patientsList)
+		if limit < 1 {
+			break
+		}
+	}
+
+	if limit > 0 {
+		selector := bson.M{
+			"_id":      bson.M{"$nin": exclusions},
+			"clinicId": clinicObjId,
+			"tags":     bson.M{"$all": tags},
+			"summary.cgmStats.dates.lastUploadDate": bson.M{
+				"$gt":  params.CgmLastUploadDateFrom,
+				"$lte": params.CgmLastUploadDateTo,
+			},
+		}
+
+		opts := options.Find()
+		opts.SetLimit(int64(limit))
+
+		opts.SetSort(bson.D{
+			{"summary.cgmStats.periods." + *params.Period + ".timeInTargetPercent", -1},
```

Comment:
I know this is a "catchall" if it hasn't retrieved up to the limit of records yet, but is there a reason why this one sorts -1 but the struct in `categories` with `Field: "timeInTargetPercent"` sorts ascending?

Reply:
confirmed matches spec